aptos-move/framework/src/extended_checks.rs

```diff
@@ -479,8 +489,65 @@ impl<'a> ExtendedChecker<'a> {
                 .fun_attributes
                 .entry(fun.get_simple_name_string().to_string())
                 .or_default()
-                .push(KnownAttribute::randomness());
+                .push(KnownAttribute::randomness(randomness_annotation.max_gas));
+        }
+    }
+
+    fn get_randomness_max_gas_declaration(
+        &self,
+        fun: &FunctionEnv,
+    ) -> Result<Option<RandomnessAnnotation>, String> {
+        for attr in fun.get_attributes() {
+            let Attribute::Apply(_, name, sub_attrs) = attr else {
+                continue;
+            };
+            if self.env.symbol_pool().string(*name).as_str() != RANDOMNESS_ATTRIBUTE {
+                continue;
+            }
+            if sub_attrs.len() >= 2 {
+                return Err(
+                    "Randomness attribute should only have one `max_gas` property.".to_string(),
+                );
+            }
+            if let Some(sub_attr) = sub_attrs.first() {
+                let Attribute::Assign(_, key_symbol, attr_val) = sub_attr else {
+                    return Err(
+                        "Randomness attribute should only have one `max_gas` property.".to_string(),
+                    );
+                };
+                let key = self.env.symbol_pool().string(*key_symbol);
+                if key.as_str() != RANDOMNESS_MAX_GAS_CLAIM {
+                    return Err(format!(
+                        "Unknown key for randomness attribute: {}",
+                        key.as_str()
+                    ));
+                }
+
+                let msg =
+                    "Property `max_gas` should be a positive integer less than 2^64".to_string();
```

Comment:
it probably should be less than the maximum gas we allowed, as least in the error message

Reply:
but that limit is an on-chain config that's hard to know at compile time?

Reply:
yes, that's why I only suggested changing error message lol

Reply:
what's the exact error msg in your mind?

Reply:
less than maximum_number_of_gas_units allowed?

Reply:
how would dev figure out `maximum_number_of_gas_units`?

Reply:
I don't know, we can also just put `aptos_global_constants::MAX_GAS_AMOUNT` here, but anyway it's weird since it only errors out when it exceeds u64::max

Reply:
if dev specifies u64::max + 1, it's surely won't work because we use u64 to represent gas amount throughout the codebase.

If dev specifies u64::max - 1, it's technically not sure if it will work, depending on on-chain gas params. So it should be a runtime error?